zerver/tests/test_service_bot_system.py

```diff
@@ -413,20 +414,20 @@ def test_bot_send_pm_with_empty_recipients_list(self) -> None:
             self.bot_handler.send_message(message={"type": "private", "to": []})
 
 
-FuncT = TypeVar("FuncT", bound=Callable[..., None])
+ParamT = ParamSpec("ParamT")
 
 
-def for_all_bot_types(test_func: FuncT) -> FuncT:
+def for_all_bot_types(test_func: Callable[ParamT, None]) -> Callable[ParamT, None]:
     @wraps(test_func)
-    def _wrapped(*args: object, **kwargs: object) -> None:
+    def _wrapped(*args: ParamT.args, **kwargs: ParamT.kwargs) -> None:
         assert len(args) > 0
         self = cast(TestServiceBotEventTriggers, args[0])
```

Comment:
A better type using `Concatenate` would let us fix this `cast`.

Reply:
Fixed. One thing to note here is that `TestServiceBotEventTriggers` is currently treated as `Any` so incorrect usage of this decorator won't be reported. This will change when django-stubs gets integrated.